domain/credential/service/service.go

```diff
@@ -124,3 +191,183 @@ func (s *Service) WatchCredential(ctx context.Context, tag names.CloudCredential
 	}
 	return nil, errors.NotYetAvailablef("credential watcher")
 }
+
+func (s *Service) modelsUsingCredential(ctx context.Context, id credential.ID) (map[model.UUID]string, error) {
+	models, err := s.st.ModelsUsingCloudCredential(ctx, id)
+	if err != nil && !errors.Is(err, errors.NotFound) {
+		return nil, errors.Trace(err)
+	}
+	return models, nil
+}
+
+func (s *Service) validateCredentialForModel(modelUUID model.UUID, cld cloud.Cloud, id credential.ID, cred *cloud.Credential) []error {
+	ctx, err := s.validationContextGetter(modelUUID)
```

Comment:
The check is already done by the one and only method which calls this. It's done there because there's other set up stuff which is doesn't make sense to do if this is nil. I can duplicate the check but seems pointless. Maybe a comment will suffice.

Reply:
I changed the signature and added a (duplicate) check